src/main/java/net/neoforged/neoforge/fluids/FluidUnitSpec.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.fluids;
+
+import com.mojang.datafixers.util.Either;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.DataResult;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import net.minecraft.network.chat.Component;
+import net.minecraft.util.Mth;
+import org.jetbrains.annotations.Nullable;
+
+public class FluidUnitSpec {
```

Comment:
For example sake, and to prevent issues with conflicting definitions, Neo should define fluid units for water, lava, and milk.

Lava and milk would probably be just buckets, which would mostly prevent anyone else from doing so (making it conflict in dev instead of waiting to conflict at runtime when someone else tries to do it). Water I feel is worth showing bottles (at the 250mb standard) in the tooltip, but if people would rather leave it to just buckets that would be a reasonable option.

Probably should also do a test mod which defines more units to a fluid, perhaps do an example with the metal standards.